pkg/catalog/operations/catalog.go

```diff
@@ -258,12 +258,27 @@ func (c *Catalog) List(targetNamespace string) error {
 	response, err := client.List(ctx, &grpc_catalog_go.ListApplicationsRequest{
 		Namespace: targetNamespace,
 	})
+
+	if searchString != "" {
+		filter(response, searchString)
+	}
+
 	if err != nil {
 		return c.ResultPrinter.PrintResultOrError(nil, nerrors.FromGRPC(err))
 	}
 	return c.ResultPrinter.PrintResultOrError(response, nil)
 }
 
+func filter(response *grpc_catalog_go.ApplicationList, searchString string) {
```

Comment:
Let's rename the function so that it is cleared what is the expected functionality.

```suggestion
func filterByName(response *grpc_catalog_go.ApplicationList, searchString string) {
```

Reply:
Yes it is clearer. 
I have changed it according to your suggestion.